src/cvar.ml

```diff
@@ -27,14 +27,15 @@ struct
   end
 
   let eval context t0 =
-    let res = ref Field.zero in
+    let open Field in
+    let res = of_int 0 in
     let rec go scale = function
-      | Constant c -> res := Field.add !res (Field.mul scale c)
-      | Var v -> res := Field.add !res (Field.mul scale (context v))
-      | Scale (s, t) -> go (Field.mul s scale) t
+      | Constant c -> res += mul scale c
```

Comment:
It also might be worth having one field element that we copy the constant `c` into and then do a mutable multiply with `scale`

Reply:
@imeckler Won't this lead to the same number of allocations as currently?

Reply:
I guess I was thinking just have one constant that you allocate outside the loop like
```
let c0 = Field.of_int 0 in
let rec go = ...
  | Constant c ->
    Field.copy ~src:c ~dst:c0;
    c0 *= scale;
    res += c0
```